### ISSUE TITLE
Turn down upstreamed/obsolete tools

### ISSUE THREAD

**Issue by myk002** [DFHack/dfhack#2474]
- autogems, gui/autogems (though rock crystal saving is still needed)
- dwarfmonitor misery widget, date widget, weather widget
- ~~follow~~ (scripted following is still useful)
- ~~mousequery~~ (could morph into an extended version of the vanilla "mousequery-like" widget)
- ~~search~~ (many screens still need search, and search bars should get keyboard focus automatically)
- ~~tidlers~~ (people actually really want this back)
- tweak (various)
- devel/find-twbt
- fix (various)
- gui/settings-manager

Any others? Are there parts of the tools named above that we should keep?

**Comment by ab9rf** [comment]
i think it's premature to declare anything obsolete until we get a much closer look at 50.01 and at what the community wants

**Comment by myk002** [comment]
The list certainly needs verification. It's intended as a gathering of prime suspects to investigate first.

**Comment by StevenLaczko** [comment]
I'm new here, but assuming "search" is what I think it is, we drill need it. Many parts of the GUI still need a search functionality in v50.

**Comment by myk002** [comment]
Good point. I'm also seeing a need to automatically set keyboard focus to the search bars where they do exist in vanilla DF. I'll remove `search` from the candidates list

**Comment by lethosor** [comment]
regarding `tidlers`: if DF removed the feature that the script `t`oggles, we would probably want to replace it with an overlay or similar with a different name.

**Comment by lethosor** [comment]
`follow` could _potentially_ be replaced with a one-liner that sets the actively-followed unit, if there is a desire for a way to follow via command. (Not sure it's really more convenient, but it would allow you to go through the DFHack hotkey system if for some reason you wanted to rebind unit following separately from DF.)

**Comment by ab9rf** [comment]
i can totally see a streamer wanting there to be a scriptable way to select a specific unit for following

**Comment by Moth-Tolias** [comment]
`fix/population-cap` *might* be on the shortlist; migrant behaviour has definitely changed
`uniform-unstick` seemingly has a button dedicated to it now? though we're not sure it quite works
maybe `break-dance` - has anyone encountered that bug recently?
`digv` has been obsolete for some time in our opinion what with the native automining designation, but the other dig commands are still relevant, so it probably wouldn't be worth the effort to remove.
is there any point in keeping `binpatch` around?
`modtools/extra-gamelog`: depending on how things go we might need to reimplement the gamelog almost from scratch.

**Comment by lethosor** [comment]
`digv` is still useful in cases where you want to see _where_ the vein is before it's dug.

`binpatch`: potentially - not used a lot these days, but the capability is useful, and it's one of the most stable tools we have because it doesn't depend on DF.

**Comment by myk002** [comment]
`fix-blood-del` didn't have any effect on my 50.05 savegame. has this issue been fixed?

**Comment by Moth-Tolias** [comment]
`fix/dead-units` is due to be upstreamed.

**Comment by myk002** [comment]
the cave adaptation mechanic appears to have been removed. this means we might be able to turn down `adaptation`

**Comment by Moth-Tolias** [comment]
from what i understood that was a bug rather than an intentional removal

**Comment by cppcooper** [comment]
> `follow` could _potentially_ be replaced with a one-liner that sets the actively-followed unit

I think it's safe to say that such a command would be based on names/nicknames/role (role as in profession, current job, or active labors)

**Comment by lethosor** [comment]
Why is autogems on this list? Did gem cutting change meaningfully in v50? There doesn't seem to be anything under the vanilla work orders menu for automating gem cutting. Yeah, you can make work orders manually for specific gems, but autogems handles that for you.

![image](https://user-images.githubusercontent.com/3719547/215678205-d4e825ff-856f-4b28-a60a-5aed89f400ef.png)

**Comment by myk002** [comment]
It was my understanding that you could now specify "cut gems" in the manager orders. Autogems might still have a place in giving more control, but it is no longer otherwise required to add orders for every gem type
